txscript/engine.go

```diff
@@ -243,10 +242,17 @@ func (vm *Engine) curPC() (script int, off int, err error) {
 	return vm.scriptIdx, vm.scriptOff, nil
 }
 
+// isWitnessVersionActive returns true if a witness program was extracted
+// during the initialization of the Engine, and the program's version matches
+// the specified version.
+func (vm *Engine) isWitnessVersionActive(version uint) bool {
+	return vm.witnessProgram != nil && uint(vm.witnessVersion) == version
+}
+
 // verifyWitnessProgram validates the stored witness program using the passed
 // witness as input.
 func (vm *Engine) verifyWitnessProgram(witness [][]byte) error {
-	if vm.witnessVersion == 0 {
+	if vm.isWitnessVersionActive(0) {
```

Comment:
This looks like it breaks the intended checks.  It was correct to only check the witness version first.  Otherwise, the else case would be wrong because it could be triggered due to the witness program being nil and not because of the version being nonzero.